projectq/ops/_basics.py

```diff
@@ -262,12 +272,19 @@ def __eq__(self, other):
             return True
         return False
 
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
     def __str__(self):
-        raise NotImplementedError('This gate does not implement __str__.')
+        return("MatrixGate(" + str(self.matrix.tolist()) + ")")
```

Comment:
Why do you convert to list here? Isn't it better if it is visible in the output that self.matrix is a numpy array (i.e., no commas between elements vs. commas between elements for `list`s)?

Reply:
Just so that the `CommandPrinter` prints such a gate on one line. A numpy matrix string has line breaks which then looks slightly weird to me.

We can use numpy arrays for this string once we switch from numpy matrix to numpy arrays #287